### PR TITLE
287 events log table

### PR DESCRIPTION
**Pull request by je-lopez**

### CODE REVIEW

**Comment by je-lopez** [comment]
issue: https://github.com/LearnersGuild/curriculum/issues/287

File: database/migrations/20170922112122_create_event_log_table.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+exports.up = knex =>
+  Promise.all([
+    knex.schema.createTable('event_logs', table => {
+      table.timestamp('occurred_at').defaultTo(knex.fn.now())
+      table.string('user_id').notNullable()
+      table.string('type').notNullable()
+      table.jsonb('metadata').notNullable()
+    })
+    .then(tables =>
+      databaseUtils.migrateCheckLogs(knex)
```
Comment (deadlyicon) [inline database/migrations/20170922112122_create_event_log_table.js]: we do not need these to be in utils going forward. just define these functions at the bottom of this migration and call them here.

File: database/queries.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ const getChecksForUserAndLabels = ({userId, labels}) => {
 const getCheckLogsForUsers = userIds => {
   return knex
     .select('*')
-    .from('check_log')
+    .from('event_logs')
+    .where('type', 'check')
```
Comment (deadlyicon) [inline database/queries.js]: perfect!

File: database/utils.js [hunk 3]
```diff
@@ -18,9 +18,61 @@ const migrateSkillsDiff = (knex, diffText) => {
   )
 }
 
+const migrateCheckLogs = (knex) => {
```
Comment (deadlyicon) [inline database/utils.js]: move this method into your migration. we only need it once.

File: database/utils.js [hunk 4]
```diff
@@ -18,9 +18,61 @@ const migrateSkillsDiff = (knex, diffText) => {
...
+              }
+            })
+            .into('event_logs')
+        )
+      })
+      return Promise.all(inserts)
+    })
+}
+
+const migrateSkillChecks = (knex) => {
```
Comment (deadlyicon) [inline database/utils.js]: move this one too

**Pull request closed without merging**